container/go/cmd/puller/puller.go

```diff
@@ -46,8 +48,8 @@ var (
 
 // NOTE: This function is adapted from https://github.com/google/go-containerregistry/blob/master/pkg/crane/pull.go
 // with slight modification to take in a platform argument.
-// Pull the image with given <imgName> to destination <dstPath> with optional required platform specifications.
-func pull(imgName, dstPath string, platform v1.Platform) {
+// Pull the image with given <imgName> to destination <dstPath> with optional cache files and required platform specifications.
+func pull(imgName, dstPath, cachePath string, platform v1.Platform) {
```

Comment:
I'm seeing calls to log.Fatalf inside this function. Convention in Go is to never panic or die inside functions. Return an error here instead and handle the error in a log.Fatalf call in the main function.

Reply:
+1 I'm gonna approve and merge this one https://github.com/bazelbuild/rules_docker/pull/907/files and then you can fix all the returns in this PR.